src/main/java/com/puppycrawl/tools/checkstyle/PropertyCacheFile.java

```diff
@@ -143,28 +143,50 @@ public void load() throws IOException {
      */
     public void persist() throws IOException {
         final Path path = Paths.get(fileName);
-        Path directory = path.getParent();
+        final Path directory = path.getParent();
 
         if (directory != null) {
-            if (Files.isSymbolicLink(directory)) {
-                final Path actualDir = directory.toRealPath();
-
-                if (Files.isDirectory(actualDir)) {
-                    directory = actualDir;
-                }
-                else {
-                    throw new IOException(
-                            "Resolved symbolic link " + directory
-                                    + " is not a directory.");
-                }
-            }
-            Files.createDirectories(directory);
+            updateDirectory(directory);
```

Comment:
make it like: `OsSpecificUtil.updateDirectory(directory);`

where OsSpecificUtil is static class  in PropertyCacheFile class.

exclude by:
```
<plugin> 
    <groupId>org.jacoco</groupId>
    <artifactId>jacoco-maven-plugin</artifactId>
    <configuration>
        <excludes>
            <exclude>com/baeldung/**/ExcludedPOJO.$OsSpecificUtilclass</exclude>
```

Reply:
Done @romani